app/models/user.rb

```diff
@@ -0,0 +1,14 @@
+class User < ActiveRecord::Base
+	has_many :todos
+
+	def authenticate(password)
+		hash = BCrypt::Engine.hash_secret(password, self.password_salt)
+		self.password_hash == hash	
+	end
+
+	def encrypt_password(password)
+		self.password_salt = BCrypt::Engine.generate_salt
+		self.password_hash = BCrypt::Engine.hash_secret(password, self.password_salt)
+	end
+
+end
```

Comment:
I'm a bit stubborn with it comes to gems hhahaha. wanted to learn the rough stuff first before the good ones. I had a sample project once using devise.